src/App.js

```diff
@@ -24,10 +25,17 @@ export default class App extends Component {
       <div className="App">
 
         {/*Make a list of all your friends*/}
+        <h1> Make a list of all your friends </h1>
+        {allFriends.map(friend=><div>{`${friend.first} ${friend.last}`}</div>)}
 
         {/*Make a list of just your best friends*/}
+        <h1> Make a list of just your best friends </h1>
+        {bestFriends.map(friend=><div>{`${friend.first} ${friend.last}`}</div>)}
+
 
         {/*Show the average age of your best friends*/}
+        <h1> Show the average age of your best friends: {bestFriends.reduce((acc, {age}, i, arr)=> acc+age/arr.length, 0)} </h1>
+        {/* !!!??? is it cool to use arr.length? if I used bestfriends.length is it less pure? */}
```

Comment:
I would -always- prefer to use it the way you are using it - like you said, it would be less pure to use bestFriends.length - specifically because imagine that this function was defined somewhere else and passed in

```
import myAverager from './functional-tools';

{bestFriends.reduce(myAverager)} 
```

If it was written like this, myAverager has **no** context for 'bestFriends'.

So to reiterate, really awesome and clever catch :)